### PR TITLE
Enable goreleaser in the repo

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
This sets up goreleaser for this repo to deploy to docker and as a package for homebrew.  Try this out:

```sh
make goreleaser_test
docker run -it trussworks/find-guardduty-user:v0.0.0 version
```

You will see output like this:

```text
v0.0.0-next
```

Similarly you can try one of the build artifacts for distribution:

```sh
./dist/find-guardduty-user_darwin_amd64/find-guardduty-user help
```

### CODE REVIEW

**Comment by chrisgilmerproj** [comment]
I'm unclear how to actually do the release for this. I have set up Docker and Github env vars in the circleci project settings. I have a sense that I need to tag this as `v0.0.1` or something similar so that the release will happen. But do I tag before or after merging? This is something I could use advice on.

**Comment by chrisgilmerproj** [comment]
I'm going to pause on this for a bit and see if I can't look at a different deploy mechanism.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -20,3 +20,12 @@ repos:
...
       - id: markdownlint
+
+  - repo: local
+    hooks:
+      - id: goreleaser-check
+        name: Validate goreleaser config  yaml
+        entry: goreleaser check
+        language: system
+        files: '.goreleaser.yml'
+        pass_filenames: false
```
Comment (chrisgilmerproj) [inline .pre-commit-config.yaml:31]: This business of checking the config file on merge seems like a good practice for us. We can put it in trussworks/pre-commit-hooks if we like it.

**Comment by eeeady** [comment]
goreleaser docs suggest pushing a tag to mainline then running the utility.  How this has worked for me using CI, I would merge changes into mainline -> use the github UI to create a new release -> new release implicitly creates new tag on mainline -> new release triggers CI job specific to release (runs goreleaser) -> goreleaser interprets newest semver tag as new relase to build etc.

I hope that kinda helps clarify what it's doing.

As a nice to have, it might be good to have a check on the version noted in go before we run go releaser. maybe?

Did you do a pre-release using this config? I wanna sanitycheck the stuff it pushed into brew. We ran into some heartache cross platform builds but I'm hoping that's _just_ cuz those utilities had to integrate with a keychain.

**Comment by chrisgilmerproj** [comment]
> goreleaser docs suggest pushing a tag to mainline then running the utility. How this has worked for me using CI, I would merge changes into mainline -> use the github UI to create a new release -> new release implicitly creates new tag on mainline -> new release triggers CI job specific to release (runs goreleaser) -> goreleaser interprets newest semver tag as new relase to build etc.
> 
> I hope that kinda helps clarify what it's doing.

I think what you're advocating is that I tag `v0.0.1` on my last git hash for this PR and push that to Github so that CircleCI picks it up and runs. Do I need to do that via the Github UI? Otherwise do I need to merge in the new `release` workflow to mainline to get it to work with the tag?

> As a nice to have, it might be good to have a check on the version noted in go before we run go releaser. maybe?

You're advocating for me to build the binaries with goreleaser and then check the version matches the tag? We can probably do that via the docker check. I think I can do `goreleaser build`, test the version against the tag, and then `goreleaser` to release.

> Did you do a pre-release using this config? I wanna sanitycheck the stuff it pushed into brew. We ran into some heartache cross platform builds but I'm hoping that's _just_ cuz those utilities had to integrate with a keychain.

I have not done a pre-release yet. Are you suggesting that I do this from CircleCI or locally? This utility doesn't have any dependencies on osx keychain or the aws-vault storage backend so I think cross-compiling should be fine.

**Comment by chrisgilmerproj** [comment]
I tagged this branch as `v0.0.1` and did a release manually from my machine. It updated homebrew and it was easily installable and worked:

```sh
$ find-guardduty-user help
Find Users that triggered GuardDuty findings

Usage:
  find-guardduty-user [command]

Available Commands:
  completion  Generates bash completion scripts
  find        Find Users that triggered GuardDuty findings
  help        Help about any command
  version     Print the version

Flags:
  -h, --help   help for find-guardduty-user

Use "find-guardduty-user [command] --help" for more information about a command.
$ find-guardduty-user version
0.0.1
```

Here's the release: https://github.com/trussworks/homebrew-tap/blob/master/find-guardduty-user.rb

**Comment by chrisgilmerproj** [comment]
Here's the release on Github: https://github.com/trussworks/find-guardduty-user/releases

**Comment by chrisgilmerproj** [comment]
I see the release code happening in circleci is stuck because I released already, which is fine. I believe the only thing not tested here is the dockerhub integration.

**Comment by eeeady** [comment]
> > goreleaser docs suggest pushing a tag to mainline then running the utility. How this has worked for me using CI, I would merge changes into mainline -> use the github UI to create a new release -> new release implicitly creates new tag on mainline -> new release triggers CI job specific to release (runs goreleaser) -> goreleaser interprets newest semver tag as new relase to build etc.
> > I hope that kinda helps clarify what it's doing.
> 
> I think what you're advocating is that I tag `v0.0.1` on my last git hash for this PR and push that to Github so that CircleCI picks it up and runs. Do I need to do that via the Github UI? Otherwise do I need to merge in the new `release` workflow to mainline to get it to work with the tag?
> 
yes. you can do that either in the `release` workflow on the github UI or pushing a tag. You don't necessarily need to merge... if you release from the ui off of the working branch it should run as is in your PR

> > As a nice to have, it might be good to have a check on the version noted in go before we run go releaser. maybe?
> 
> You're advocating for me to build the binaries with goreleaser and then check the version matches the tag? We can probably do that via the docker check. I think I can do `goreleaser build`, test the version against the tag, and then `goreleaser` to release.
> 

No actually... I think it would be nice in the circleci release job to validate that the build it's going to do will have matching tags and version strings.

> > Did you do a pre-release using this config? I wanna sanitycheck the stuff it pushed into brew. We ran into some heartache cross platform builds but I'm hoping that's _just_ cuz those utilities had to integrate with a keychain.
> 
> I have not done a pre-release yet. Are you suggesting that I do this from CircleCI or locally? This utility doesn't have any dependencies on osx keychain or the aws-vault storage backend so I think cross-compiling should be fine.

I was confused you can try it either way... I think on CircleCi gets more complicated and if it works locally you might wanna check it on Circle.

**Comment by eeeady** [comment]
Looks like that worked btw. I was able to brew install and run the version command.

**Comment by eeeady** [comment]
LMK if you need help with the docker hub stuff... I _think_ I documented it in engineering playbook here: https://github.com/trussworks/Engineering-Playbook/blob/master/developing/command-line-tools/HOW2GORELEASER.md#create-a-docker-repo

**Comment by chrisgilmerproj** [comment]
So it turns out the Dockerhub stuff didn't release because the github release blocked it. I'll try to push that out manually and then get this in.

**Comment by chrisgilmerproj** [comment]
Dockerhub is now updated: https://hub.docker.com/repository/docker/trussworks/find-guardduty-user/general

```sh
docker pull trussworks/find-guardduty-user:v0.0.1
docker run -it trussworks/find-guardduty-user:v0.0.1 version
0.0.1
```

**Pull request merged**